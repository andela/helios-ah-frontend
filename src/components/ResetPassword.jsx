import React, { Component } from 'react';
// import * as passwordResetImage from '../image/resetPassword.svg';
import * as axios from '../utilities/apiRequests'

export class ResetPassword extends Component {
  state = {
    email: null,
    status: null
  };

  onSubmit = async (e) => {
    e.preventDefault();
    const data = {
      email: this.state.email
    }
    const status = this.state.status !== 'success'
      ? await axios.Post('/user/requests/password/reset', data)
      : null;

    this.setState({ status: status.status });

    this.state.status === 'error'
      ? this.statusHandler('User Not Found')
      : this.statusHandler('Password reset link has been sent to your mail');
  }

  onChange = (e) => {
    this.clearStatus();
    let value = e.target.value.trim();
    this.setState({ email: value });
  }

  statusHandler = (message = '') => {
  this.refs.status.innerHTML = message;
    this.clearEmail();
  }
  clearStatus(e){
   this.refs.status.innerHTML ='';
  }
  clearEmail(e){
    this.refs.email.value ='';
  }
  render() {
    return (
      <div className='container passwordResetBody'>
        <div className='row passwordResetBody'>
          <div className='d-none d-sm-block col-sm-4 col-md-6 passwordResetImageDiv'>
            <span className='passwordResetImage'/>
          </div>
          <div className='col-sm-8 col-md-6 text-center textWithInput'>
            <div className='resetPasswordText'>
              <h1 className='passwordResetH1'>FORGOT YOUR PASSWORD?</h1>
              <h3 className='passwordResetH3'>ENTER YOUR EMAIL ADDRESS BELOW AND REQUEST A RESET LINK</h3>
            </div>
            <div>
              <form onSubmit={this.onSubmit} className='resetPasswordForm'>
                <div className='passwordResetInputDiv'>
                  <input type='text' className='passwordResetInput' id='email' name='email' ref="email" placeholder='Enter your email here' onChange={this.onChange} />
                </div>
                <span className='passwordResetInfo' ref="status"></span>
                <div className='passwordresetbuttonDiv'>
                  <button className='passwordResetButton'>REQUEST RESET LINK</button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default ResetPassword;