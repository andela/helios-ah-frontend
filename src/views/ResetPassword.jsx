import React, { Component } from 'react';
import * as axios from '../utilities/apiRequests';
import ResetPasswordForm from '../components/ResetPasswordForm';

class ResetPassword extends Component {
  constructor(props) {
    super(props);
    this.spanRef = React.createRef();
    this.emailRef = React.createRef();
    this.state = {
      email: "null",
      status: "null",
    };
  }

  onSubmit = async (e) => {
    e.preventDefault();
    const { status, email } = this.state;

    const data = {
      email
    };

    const currentStatus = (status !== 'success' || status !== undefined)
      ? await axios.Post('/user/requests/password/reset', data)
      : null;

    this.setState({ status: currentStatus.status });
    const newStatus = this.state;

    // eslint-disable-next-line no-nested-ternary
    return (newStatus.status === "null" || newStatus.status === undefined)
      ? this.statusHandler('Connection Error, Please try again Later')
      : (newStatus.status === 'error' || newStatus.status === undefined)
        ? this.statusHandler('User Not Found')
        : this.statusHandler('Password reset link has been sent to your mail');
  }


  onChange = (e) => {
    this.clearStatus();
    const value = e.target.value.trim();
    this.setState({ email: value });
  }

  statusHandler = (message = '') => {
    this.spanRef.current.innerHTML = message;
    this.clearEmail();
  }

  clearStatus() {
    this.spanRef.current.innerHTML = '';
  }

  clearEmail() {
    this.emailRef.current.value = '';
  }

  render() {
    return (
      <ResetPasswordForm
        onChange={this.onChange}
        onSubmit={this.onSubmit}
        emailRef={this.emailRef}
        spanRef={this.spanRef}
      />
    );
  }
}

export default ResetPassword;
