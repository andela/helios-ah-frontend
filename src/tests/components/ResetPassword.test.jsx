import React from 'react';
import { shallow } from 'enzyme';
import ResetPassword from '../../components/ResetPasswordForm';

const propsObject = {
  onChange: () => (''),
  onSubmit: () => (''),
  onClick: () => (''),
  emailRef: {},
  spanRef: {},
  buttonRef: {}
};

describe('Test to successfully render <ResetPassword />', () => {
  test('renders the <ResetPassword /> components', () => {
    const wrapper = shallow(<ResetPassword {...propsObject} />);
    expect(wrapper.length).toBe(1);
  });

  test('should render Passowrd reset page', () => {
    const wrapper = shallow(<ResetPassword {...propsObject} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('expect the element h1 to have predefined content', () => {
    const wrapper = shallow(<ResetPassword {...propsObject} />);
    expect(wrapper.find('h1').text()).toBe('FORGOT YOUR PASSWORD?');
  });

  test('expect the element h3 to have a ascribed content', () => {
    const wrapper = shallow(<ResetPassword {...propsObject} />);
    expect(wrapper.find('h3').text())
      .toBe('ENTER YOUR EMAIL ADDRESS BELOW AND REQUEST A RESET LINK');
  });
});
