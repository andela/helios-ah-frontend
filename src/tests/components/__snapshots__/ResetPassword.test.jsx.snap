// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to successfully render <ResetPassword /> should render Passowrd reset page 1`] = `
<div
  className="container passwordResetBody"
>
  <div
    className="row passwordResetBody"
  >
    <div
      className="d-none d-sm-block col-sm-4 col-md-6 passwordResetImageDiv"
    >
      <span
        className="passwordResetImage"
      />
    </div>
    <div
      className="col-sm-8 col-md-6 text-center textWithInput"
    >
      <div
        className="resetPasswordText"
      >
        <h1
          className="passwordResetH1"
        >
          FORGOT YOUR PASSWORD?
        </h1>
        <h3
          className="passwordResetH3"
        >
          ENTER YOUR EMAIL ADDRESS BELOW AND REQUEST A RESET LINK
        </h3>
      </div>
      <div>
        <form
          className="resetPasswordForm"
          onSubmit={[Function]}
        >
          <div
            className="passwordResetInputDiv"
          >
            <FormInput
              Ref={Object {}}
              className="passwordResetInput"
              id="email"
              name="email"
              onChange={[Function]}
              placeHolder="Enter your email here"
              type="text"
            />
          </div>
          <span
            className="passwordResetInfo"
          />
          <div
            className="passwordresetbuttonDiv"
          >
            <SubmitButton
              className="passwordResetButton"
              columnAttribute={null}
              id="resetPassword"
              isRequestSent={false}
              onClick={[Function]}
              type="button"
              value="REQUEST RESET LINK"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
